### ISSUE TITLE
Volume is working but doesn't update.

### ISSUE THREAD

**Issue by MrBrownFromLa** [grzegorz914/homebridge-denon-tv#162]
Hi,

First of all I would like to thank you for this Homebridge plugin: so great!
I own a Marantz M-CR611, all work great, but the volume control: if I update the volume with the slider it's updating but if I change the volume with an automation or button, the slider doesn't update volume in return.
Could you help me find the problem?
Thanks for your time.

here is my config. tell me if you need more info...

>     "devices": [
        {
            "name": "Marantz M-CR611",
            "host": "192.168.1.25",
            "port": 80,
            "zoneControl": 0,
            "volumeControl": 2,
            "masterPower": true,
            "sensorPower": true,
            "masterVolume": true,
            "sensorVolume": false,
            "masterMute": true,
            "sensorMute": false,
            "infoButtonCommand": "MNINF",
            "disableLogInfo": false,
            "disableLogDeviceInfo": false,
            "enableDebugMode": false,
            "getInputsFromDevice": false,
            "inputs": [
                {
                    "name": "Digital In",
                    "reference": "DIGITALIN1",
                    "mode": "SI",
                    "switch": true,
                    "displayType": 2
                },
                {
                    "name": "Bluetooth",
                    "reference": "BLUETOOTH",
                    "mode": "SI",
                    "switch": true,
                    "displayType": 2
                },
                {
                    "name": "iRadio",
                    "reference": "NET",
                    "mode": "SI",
                    "switch": true,
                    "displayType": 2
                },
                {
                    "name": "France Inter",
                    "reference": " 01",
                    "mode": "FV"
                },
                {
                    "name": "FIP",
                    "reference": " 02",
                    "mode": "FV"
                },
                {
                    "name": "France Musique",
                    "reference": " 03",
                    "mode": "FV"
                },
                {
                    "name": "France Culture",
                    "reference": " 04",
                    "mode": "FV"
                },
                {
                    "name": "TSF Jazz",
                    "reference": " 05",
                    "mode": "FV"
                },
                {
                    "name": "Radio Nova",
                    "reference": " 06",
                    "mode": "FV"
                },
                {
                    "name": "Le Djam",
                    "reference": " 07",
                    "mode": "FV"
                },
                {
                    "name": "Radio Meuh",
                    "reference": " 08",
                    "mode": "FV"
                },
                {
                    "name": "Radio Sympa",
                    "reference": " 09",
                    "mode": "FV"
                },
                {
                    "name": "Le Gri-Gri",
                    "reference": " 10",
                    "mode": "FV"
                }
            ],
            "getSurroundsFromDevice": false,
            "buttonsMainZone": [
                {
                    "name": "Vol +",
                    "reference": "MVUP",
                    "displayType": 0
                },
                {
                    "name": "Vol -",
                    "reference": "MVDOWN",
                    "displayType": 0
                }
            ],
            "enableMqtt": false,
            "mqttAuth": false,
            "mqttDebug": false
        }
    ],

**Comment by grzegorz914** [comment]
In normal case need to be updated automatically if volume changed on AVR side, just checked and working OK.
If you can please enable debug mode in plugin settings and get the log during this operation.

**Comment by MrBrownFromLa** [comment]
If I change the volume with the Amplifier button or with a dummy switch (who send the "MVUP" command)
nothing change in the debug log...
> 
[24/12/2022 14:55:47] [homebridge-denon-tv] Device: 192.168.1.25 Marantz M-CR611, debug: State: {
  "Power": [
    {
      "value": [
        "ON"
      ]
    }
  ],
  "InputFuncSelect": [
    {
      "value": [
        "NET"
      ]
    }
  ],
  "MasterVolume": [
    {
      "value": [
        "-66.0"
      ]
    }
  ],
  "Mute": [
    {
      "value": [
        "off"
      ]
    }
  ]
}

[24/12/2022 14:55:52] [homebridge-denon-tv] Device: 192.168.1.25 Marantz M-CR611, debug: State: {
  "Power": [
    {
      "value": [
        "ON"
      ]
    }
  ],
  "InputFuncSelect": [
    {
      "value": [
        "NET"
      ]
    }
  ],
  "MasterVolume": [
    {
      "value": [
        "-66.0"
      ]
    }
  ],
  "Mute": [
    {
      "value": [
        "off"
      ]
    }
  ]
}

If I use the volume slider inside the home app it works.

> 
[24/12/2022 15:03:42] [homebridge-denon-tv] Device: 192.168.1.25 Marantz M-CR611, set new Mute state successful, state: OFF
[24/12/2022 15:03:42] [homebridge-denon-tv] Device: 192.168.1.25 Marantz M-CR611, set new Volume level successful, volume: -59 dB
[24/12/2022 15:03:42] [homebridge-denon-tv] Device: 192.168.1.25 Marantz M-CR611, debug: State: {
  "Power": [
    {
      "value": [
        "ON"
      ]
    }
  ],
  "InputFuncSelect": [
    {
      "value": [
        "NET"
      ]
    }
  ],
  "MasterVolume": [
    {
      "value": [
        "-60.0"
      ]
    }
  ],
  "Mute": [
    {
      "value": [
        "off"
      ]
    }
  ]
}

**Comment by grzegorz914** [comment]
All is ok here, volume changed from -66 to - 60, the slider need to be updated too.

**Comment by grzegorz914** [comment]
All is ok here, volume changed from -66 to - 60, the slider need to be updated too.

**Comment by MrBrownFromLa** [comment]
If I move volume with the slider (volume changed from -66 to - 60 in the last log above log @15:03:42), it's updated in Home app -> this work great.

But not with anything else (volume stays à -66 although I've changed it with my amp remote, button and http dummyswitch (sending MVUP) (see the first two log shows it: volume stay at -66 log @14:55:47->14:55:52), it don't seem to see the change in volume...
Thanks for your help!

**Comment by grzegorz914** [comment]
As I see You use "masterVolume": true, probably your RC or other web control device changed Zone Volume not the master.

**Comment by MrBrownFromLa** [comment]
Ok, I've got to investigate more on that! 

I've already tried changing to zone 2 also with MasterVolume: false, and even with volumeControl: false, but still volume won't refresh and in the log it don't seem to see the change if don't use the volume slider.
I also try a dummyhttpswitch sending via telnet a "MVUP"/"MVDOWN" command: the amp catch it but Homebridge won't update the change...
Anyway, it works with the slider! 

Also, in find the control protocol from Denon on a forum and if I send "SIIRADIO" command it change the input on amp but it won't update as a sensor. But if I use "SINET" (although it doesn't seem to exist in the control protocol) it works. I don't get why it act that way...
Anyway again, it works with "SINET"!
I'm just trying to understand how it works ^^

Thank you for your kind help. I'll tell you if I find something else.

[M-CR511_611_PROTOCOL_Ver1_0_0.pdf](https://github.com/grzegorz914/homebridge-denon-tv/files/10307645/M-CR511_611_PROTOCOL_Ver1_0_0.pdf)
